### PR TITLE
PR: Implement support for "Display P3" colourspace.

### PR DESCRIPTION
**Pull request by sobotka**
Requires citation adding. Used "Apple2019" as a
placeholder here. Likely requires some scrutiny from
@KelSolaar as usual.

### CODE REVIEW

File: colour/models/rgb/datasets/display_p3.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    documentation/coregraphics/cgcolorspace/1408916-displayp3#discussion
+"""
+
+from __future__ import division, unicode_literals
+
+import numpy as np
+from functools import partial
+
+from colour.colorimetry import ILLUMINANTS
+from colour.models.rgb import (RGB_Colourspace, gamma_function,
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: ['colour.models.rgb.gamma_function' imported but unused](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381&bid=15649171)

File: colour/models/rgb/datasets/display_p3.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+References
+----------
+-   :cite:`Apple2019` : Apple. (2019). https://developer.apple.com/\
+    documentation/coregraphics/cgcolorspace/1408916-displayp3#discussion
+"""
+
+from __future__ import division, unicode_literals
+
+import numpy as np
+from functools import partial
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: [Unused partial imported from functools](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/__init__.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from .color_match_rgb import COLOR_MATCH_RGB_COLOURSPACE
 from .dcdm_xyz import DCDM_XYZ_COLOURSPACE
 from .dci_p3 import DCI_P3_COLOURSPACE, DCI_P3_P_COLOURSPACE
+from .display_p3 import DISPLAY_P3_COLOURSPACE, DISPLAY_P3_COLOURSPACE
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/__init__.py]: Codacy Issue found: [Reimport 'DISPLAY_P3_COLOURSPACE' (imported line 21)](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    'Display P3',
+    DISPLAY_P3_PRIMARIES,
+    DISPLAY_P3_WHITEPOINT,
+    DISPLAY_P3_WHITEPOINT_NAME,
+    DISPLAY_P3_TO_XYZ_MATRIX,
+    XYZ_TO_DISPLAY_P3_MATRIX,
+    eotf_inverse_sRGB,
+    eotf_sRGB,
+)
+DCI_P3_COLOURSPACE.__doc__ = """
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: [undefined name 'DCI_P3_COLOURSPACE'](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py (hunk 1 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: [Unused gamma_function imported from colour.models.rgb](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/__init__.py (hunk 3 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/__init__.py]: Codacy Issue found: [redefinition of unused 'DISPLAY_P3_COLOURSPACE' from line 21](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py (hunk 2 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: ['functools.partial' imported but unused](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381&bid=15649171)

File: colour/models/rgb/datasets/__init__.py (hunk 3 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/__init__.py]: Codacy Issue found: [Reimport 'DISPLAY_P3_COLOURSPACE' (imported line 21)](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+References
+----------
+-   :cite:`Apple2019` : Apple. (2019). https://developer.apple.com/\
+    documentation/coregraphics/cgcolorspace/1408916-displayp3#discussion
+"""
+
+from __future__ import division, unicode_literals
+
+import numpy as np
+from functools import partial
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: [Unused partial imported from functools](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    'Display P3',
+    DISPLAY_P3_PRIMARIES,
+    DISPLAY_P3_WHITEPOINT,
+    DISPLAY_P3_WHITEPOINT_NAME,
+    DISPLAY_P3_TO_XYZ_MATRIX,
+    XYZ_TO_DISPLAY_P3_MATRIX,
+    eotf_inverse_sRGB,
+    eotf_sRGB,
+)
+DCI_P3_COLOURSPACE.__doc__ = """
```
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: [undefined name 'DCI_P3_COLOURSPACE'](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/display_p3.py (hunk 5 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/display_p3.py]: Codacy Issue found: ['functools.partial' imported but unused](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

File: colour/models/rgb/datasets/__init__.py (hunk 3 above)
Comment (KelSolaar) [inline colour/models/rgb/datasets/__init__.py]: Codacy Issue found: [redefinition of unused 'DISPLAY_P3_COLOURSPACE' from line 21](https://app.codacy.com/app/colour-science/colour/pullRequest?prid=4677381)

**Comment by KelSolaar** [comment]
Thanks @sobotka! Only a few small minor stylistic comments, besides that LGTM!

**Comment by KelSolaar** [comment]
Ah, another one that came, is it possible to change the email addresses to `colour-developers@colour-science.org` please?

**Comment by sobotka** [comment]
Sure.

I appear to be too thick to see any comments on the stylistic things in the code. Did you put some in?

**Comment by sobotka** [comment]
Not pedantry at all. As you know, I'm a stickler on those fronts as well.

Let me know if you have other changes.

**Comment by KelSolaar** [comment]
Thanks @sobotka! 🍾

**Pull request merged**